code/createDeviceSettingsTable.m

```diff
@@ -0,0 +1,757 @@
+                % Create new row in deviceSettingsTable and populate with metadata
+                HostUnixTime = currentSettings.RecordInfo.HostUnixTime;
+                TD_SettingsTable.action{entryNumber_TD} = sprintf('Stop Sense TD %d',streamStopCounter_TD);
+                TD_SettingsTable.recNum(entryNumber_TD) = streamStopCounter_TD;
+                TD_SettingsTable.time{entryNumber_TD} = HostUnixTime;
+                
+                % Fill in most recent time domain data settings
+                for iChan = 1:4
+                    fieldName = sprintf('chan%d',iChan);
+                    TD_SettingsTable.(fieldName){entryNumber_TD} = TDsettings(iChan).chanFullStr;
+                end
+                TD_SettingsTable.tdDataStruc{entryNumber_TD} = TDsettings;
+                inStream_TD = 0;
+                streamStopCounter_TD = streamStopCounter_TD + 1;
+                entryNumber_TD = entryNumber_TD + 1;
+            end
+        end
+        
+        % POWER DOMAIN
+        if inStream_Power && isfield(currentSettings.SenseState,'state')
+            senseState = dec2bin(currentSettings.SenseState.state,4);
+            % Check starting/stopping of time domain streaming. See
+            % documentation enum Medtronic.NeuroStim.Olympus.DataTypes.Sensing.SenseStates : byte
+            % for more details about binary number coding
+            % Same code for all streams to indicate sense off
+            if strcmp(senseState(4),'0') % Time domain streaming is off
+                % Create new row in deviceSettingsTable and populate with metadata
+                HostUnixTime = currentSettings.RecordInfo.HostUnixTime;
+                Power_SettingsTable.action{entryNumber_Power} = sprintf('Stop Sense Power %d',streamStopCounter_Power);
+                Power_SettingsTable.recNum(entryNumber_Power) = streamStopCounter_Power;
+                Power_SettingsTable.time{entryNumber_Power} = HostUnixTime;
+                
+                % Fill in most recent power domain settings
+                Power_SettingsTable.powerBands{entryNumber_Power} = powerChannels;
+                for iChan = 1:4
+                    TDsampleRates(iChan) = str2double(TDsettings(iChan).sampleRate(1:end-2));
+                end
+                TDsampleRates = unique(TDsampleRates);
+                currentTDsampleRate = TDsampleRates(~isnan(TDsampleRates));
+                Power_SettingsTable.TDsampleRates{entryNumber_Power} = currentTDsampleRate;
+                Power_SettingsTable.fftConfig(entryNumber_Power) = fftConfig;
+                
+                inStream_Power = 0;
+                streamStopCounter_Power = streamStopCounter_Power + 1;
+                entryNumber_Power = entryNumber_Power + 1;
+            end
+            
+        end
+        
+        % FFT
+        if inStream_FFT && isfield(currentSettings.SenseState,'state')
+            senseState = dec2bin(currentSettings.SenseState.state,4);
```

Comment:
Redundant with time domain check. See suggestion above.
```suggestion
```

Reply:
Moved the check of senseState to outside the loops dependent on whether streaming for TD, Power, or FFT. Updated with f6e367e